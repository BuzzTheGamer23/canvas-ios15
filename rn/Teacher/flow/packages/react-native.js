// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
