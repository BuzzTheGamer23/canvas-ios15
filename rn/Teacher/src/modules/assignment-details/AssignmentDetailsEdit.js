/**
 * @flow
 */

import React, { Component } from 'react'
import { connect } from 'react-redux'
import { updateMapStateToProps, type AssignmentDetailsProps } from './map-state-to-props'
import AssignmentActions from '../assignments/actions'
import i18n from 'format-message'
import EditSectionHeader from './components/EditSectionHeader'
import AssignmentDatesEditor from './components/AssignmentDatesEditor'
import { TextInput, Text } from '../../common/text'
import ModalActivityIndicator from '../../common/components/ModalActivityIndicator'
import { Navigation } from 'react-native-navigation'
import { ERROR_TITLE, parseErrorMessage } from '../../redux/middleware/error-handler'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import color from './../../common/colors'
import {
  View,
  StyleSheet,
  Alert,
} from 'react-native'

export class AssignmentDetailsEdit extends Component<any, AssignmentDetailsProps, any> {
  props: AssignmentDetailsProps
  state: any = {}

  static navigatorButtons = {
    rightButtons: [
      {
        title: i18n({
          default: 'Done',
          description: 'Button to close modal',
          id: 'done_edit_assignment',
        }),
        id: 'dismiss',
        testID: 'edit-assignment.dismiss-btn',
      },
    ],
    leftButtons: [
      {
        title: i18n('Cancel'),
        id: 'cancel',
        testID: 'edit-assignment.cancel-btn',
      },
    ],
  }

  constructor (props: AssignmentDetailsProps) {
    super(props)
    props.navigator.setOnNavigatorEvent(this.onNavigatorEvent)
    this.state = { assignment: Object.assign({}, props.assignmentDetails) }
  }

  componentDidMount () {
    this.props.navigator.setTitle({
      title: i18n({
        default: 'Edit Assignment Details',
        description: 'Title of Assignment details EDIT screen',
      }),
    })
  }

  renderLeftColumnLabel (text: string): React.Element<*> {
    return (<Text style={style.twoColumnRowLeftText} fontWeight={'semibold'}>{text}</Text>)
  }

  renderTextInput (fieldName: string, placeholder: string, testID: string, styleParam: Object = {}, multiline: boolean = false): React.Element<*> {
    return (
      <TextInput style={styleParam}
                 value={ this.defaultValueForInput(fieldName) }
                 multiline={ multiline }
                 placeholder={ placeholder }
                 onChangeText={ value => this.updateFromInput(fieldName, value) }
                 testID={testID}/>
    )
  }

  render (): React.Element<View> {
    let sectionTitle = i18n({
      default: 'Title',
      description: 'Assignment details edit title header',
    })
    let sectionDetails = i18n({
      default: 'Details',
      description: 'Assignment details edit details header',
    })

    let dueDatesTitle = i18n({
      default: 'Due Dates',
      description: 'Assignment details due dates header',
    })

    let savingText = i18n({
      default: 'Saving',
      description: 'Text when a request to update an assignment is made and user is waiting',
    })

    let titlePlaceHolder = i18n({ default: 'Title', description: 'Assignemnt details title placeholder' })
    let pointsPlaceHolder = i18n({ default: 'Points', description: 'Assignemnt details points placeholder' })

    return (
      <View style={{ flex: 1 }}>
        <ModalActivityIndicator text={savingText} visible={this.state.pending}/>
        <KeyboardAwareScrollView style={style.container} ref='scrollView'>

          {/* Title */}
          <EditSectionHeader title={sectionTitle} style={[style.sectionHeader, { marginTop: 0 }]}/>
          <View style={style.row}>
            { this.renderTextInput('name', titlePlaceHolder, 'titleInput', style.title, true) }
          </View>

          {/* Points */}
          <EditSectionHeader title={sectionDetails} style={style.sectionHeader}/>
          <View style={[style.row, style.twoColumnRow]}>
            { this.renderLeftColumnLabel(pointsPlaceHolder) }
            { this.renderTextInput('points_possible', pointsPlaceHolder, 'pointsInput', style.points) }
          </View>

          {/* Due Dates */}
          <EditSectionHeader title={dueDatesTitle} style={style.sectionHeader}/>
          <AssignmentDatesEditor assignment={this.props.assignmentDetails} />

        </KeyboardAwareScrollView>
      </View>
    )
  }

  updateFromInput (key: string, value: string) {
    const assignment = this.state.assignment
    assignment[key] = value
    this.setState({ assignment })
  }

  defaultValueForInput (key: string): string {
    let assignment = this.state.assignment
    return assignment[key].toString()
  }

  onNavigatorEvent = (event: NavigatorEvent) => {
    switch (event.type) {
      case 'NavBarButtonPress':
        switch (event.id) {
          case 'dismiss':
            this.actionDonePressed()
            break
          case 'cancel':
            this.actionCancelPressed()
            break
        }
        break
    }
  }

  actionDonePressed () {
    this.setState({ pending: true })
    this.props.updateAssignment(this.props.courseID, this.state.assignment, this.props.assignmentDetails)
  }

  actionCancelPressed () {
    Navigation.dismissAllModals()
  }

  componentDidUpdate () {
    if (this.state.error) {
      this.handleError()
    }
  }

  handleError () {
    setTimeout(() => { Alert.alert(ERROR_TITLE, this.state.error); delete this.state.error }, 1000)
  }

  componentWillReceiveProps (nextProps: AssignmentDetailsProps) {
    if (!nextProps.pending && nextProps.error) {
      let error = parseErrorMessage(nextProps.error.response)
      this.setState({ pending: false, error: error, assignment: Object.assign({}, this.state.assignment) })
      return
    }

    if (this.state.pending && (nextProps.assignmentDetails && !nextProps.pending)) {
      this.setState({ error: undefined })
      Navigation.dismissAllModals()
    }
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
  },
  sectionHeader: {
    marginTop: global.style.defaultPadding / 2,
  },
  row: {
    paddingTop: global.style.defaultPadding / 2,
    paddingBottom: global.style.defaultPadding / 2,
    paddingLeft: global.style.defaultPadding,
    paddingRight: global.style.defaultPadding,
  },
  twoColumnRow: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 54,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: color.grey2,
  },
  twoColumnRowLeftText: {
    flex: 1,
  },
  title: {
    height: 45,
  },
  points: {
    width: 50,
    textAlign: 'right',
  },
})

let Connected = connect(updateMapStateToProps, AssignmentActions)(AssignmentDetailsEdit)
export default (Connected: Component<any, AssignmentDetailsProps, any>)
