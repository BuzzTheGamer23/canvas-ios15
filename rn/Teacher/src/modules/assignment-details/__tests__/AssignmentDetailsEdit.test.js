/**
 * @flow
 */

import 'react-native'
import React from 'react'
import { AssignmentDetailsEdit } from '../AssignmentDetailsEdit'

const template = {
  ...require('../../../api/canvas-api/__templates__/assignments'),
  ...require('../../../api/canvas-api/__templates__/course'),
  ...require('../../../__templates__/react-native-navigation'),
}

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('../../../routing')

let course: any = template.course()
let assignment: any = template.assignment()

let defaultProps = {
  navigator: template.navigator(),
  courseID: course.courseID,
  assignmentID: assignment.assignmentID,
  refreshAssignmentDetails: (courseID: string, assignmentID: string) => {},
  assignmentDetails: assignment,
  pending: 0,
  stubSubmissionProgress: true,
}

test('renders', () => {
  let tree = renderer.create(
    <AssignmentDetailsEdit {...defaultProps} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('calls navigator.dismissModal when the done button is pressed', () => {
  let navigator = template.navigator({
    dismissModal: jest.fn(),
  })

  let tree = renderer.create(
    <AssignmentDetailsEdit {...defaultProps} navigator={navigator} />
  )

  tree.getInstance().onNavigatorEvent({
    type: 'NavBarButtonPress',
    id: 'dismiss',
  })

  expect(navigator.dismissModal).toHaveBeenCalledWith()
})
