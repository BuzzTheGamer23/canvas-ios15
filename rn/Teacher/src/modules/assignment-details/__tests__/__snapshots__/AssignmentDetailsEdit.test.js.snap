// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error occurs when done pressed 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.6)",
              "borderRadius": 12,
            },
            Object {
              "backgroundColor": "rgba(0,0,0,0.8)",
            },
            Object {
              "height": 170,
              "width": 170,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#fff",
                    "fontFamily": ".SFUIDisplay-semibold",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Saving
            </Text>
          </View>
        </View>
        <ActivityIndicator
          animating={true}
          color="#fff"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "flex": 1,
              "marginTop": -40,
            }
          }
        />
      </View>
    </View>
  </Modal>
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Array [
              Object {
                "marginTop": 8,
              },
              Object {
                "marginTop": 0,
              },
            ],
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Title
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <TextInput
          multiline={true}
          onChangeText={[Function]}
          placeholder="Title"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "fontSize": 17,
              },
              Object {
                "height": 45,
              },
            ]
          }
          testID="titleInput"
          value="learn to write code"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Details
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
            },
            Object {
              "alignItems": "center",
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          fontWeight="semibold"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Points
        </Text>
        <TextInput
          multiline={false}
          onChangeText={[Function]}
          placeholder="Points"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "fontSize": 17,
              },
              Object {
                "textAlign": "right",
                "width": 50,
              },
            ]
          }
          testID="pointsInput"
          value="20"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Due Dates
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "height": 54,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#008EE2",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Add Due Date
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "height": 24,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`modal saving is shown on assignment update 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.6)",
              "borderRadius": 12,
            },
            Object {
              "backgroundColor": "rgba(0,0,0,0.8)",
            },
            Object {
              "height": 170,
              "width": 170,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#fff",
                    "fontFamily": ".SFUIDisplay-semibold",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Saving
            </Text>
          </View>
        </View>
        <ActivityIndicator
          animating={true}
          color="#fff"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "flex": 1,
              "marginTop": -40,
            }
          }
        />
      </View>
    </View>
  </Modal>
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Array [
              Object {
                "marginTop": 8,
              },
              Object {
                "marginTop": 0,
              },
            ],
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Title
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <TextInput
          multiline={true}
          onChangeText={[Function]}
          placeholder="Title"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "fontSize": 17,
              },
              Object {
                "height": 45,
              },
            ]
          }
          testID="titleInput"
          value="learn to write code"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Details
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
            },
            Object {
              "alignItems": "center",
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          fontWeight="semibold"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Points
        </Text>
        <TextInput
          multiline={false}
          onChangeText={[Function]}
          placeholder="Points"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "fontSize": 17,
              },
              Object {
                "textAlign": "right",
                "width": 50,
              },
            ]
          }
          testID="pointsInput"
          value="20"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Due Dates
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "height": 54,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#008EE2",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Add Due Date
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "height": 24,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.6)",
              "borderRadius": 12,
            },
            Object {
              "backgroundColor": "rgba(0,0,0,0.8)",
            },
            Object {
              "height": 170,
              "width": 170,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#fff",
                    "fontFamily": ".SFUIDisplay-semibold",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Saving
            </Text>
          </View>
        </View>
        <ActivityIndicator
          animating={true}
          color="#fff"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "flex": 1,
              "marginTop": -40,
            }
          }
        />
      </View>
    </View>
  </Modal>
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Array [
              Object {
                "marginTop": 8,
              },
              Object {
                "marginTop": 0,
              },
            ],
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Title
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <TextInput
          multiline={true}
          onChangeText={[Function]}
          placeholder="Title"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "fontSize": 17,
              },
              Object {
                "height": 45,
              },
            ]
          }
          testID="titleInput"
          value="learn to write code"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Details
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
            },
            Object {
              "alignItems": "center",
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 54,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          fontWeight="semibold"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "flex": 1,
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          Points
        </Text>
        <TextInput
          multiline={false}
          onChangeText={[Function]}
          placeholder="Points"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "fontSize": 17,
              },
              Object {
                "textAlign": "right",
                "width": 50,
              },
            ]
          }
          testID="pointsInput"
          value="20"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "borderBottomColor": "#D8DBDE",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#D8DBDE",
              "borderTopWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "flex-start",
            },
            Object {
              "marginTop": 8,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontFamily": ".SFUIDisplay-semibold",
                "fontSize": 20,
              },
              Object {
                "color": "#2D3B45",
                "marginBottom": 8,
                "marginLeft": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Due Dates
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "height": 54,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#008EE2",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Add Due Date
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "height": 24,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
